.github/workflows/jira.yaml

```diff
@@ -0,0 +1,92 @@
+on:
+  issues:
+    types: [opened, closed, deleted, reopened]
+  pull_request_target:
+    types: [opened, closed, reopened]
+  issue_comment:
+    types: [created]
+  workflow_dispatch:
+
+name: Jira Sync
+
+jobs:
+  sync:
+    runs-on: ubuntu-latest
+    name: Jira sync
+    steps:    
+      - name: Login
+        uses: atlassian/gajira-login@v3
+        env:
+          JIRA_BASE_URL: ${{ secrets.JIRA_BASE_URL }}
+          JIRA_USER_EMAIL: ${{ secrets.JIRA_USER_EMAIL }}
+          JIRA_API_TOKEN: ${{ secrets.JIRA_API_TOKEN }}
+
+      - name: Set ticket type
+        id: set-ticket-type
+        run: |
+          echo "::set-output name=type::GH Issue"
+      - name: Set ticket labels
+        if: github.event.action == 'opened'
+        id: set-ticket-labels
+        run: |
+          LABELS="[consul]"
+          echo "::set-output name=labels::${LABELS}"
+          
+      - name: Check if team member
+        if: github.event.action == 'opened' && steps.set-ticket-type.outputs.type == 'Task'
+        id: is-team-member
+        run: |
+          TEAM=consul
+          ROLE="$(hub api orgs/hashicorp/teams/${TEAM}/memberships/${{ github.actor }} | jq -r '.role | select(.!=null)')"
+          if [[ -n ${ROLE} ]]; then
+            echo "Actor ${{ github.actor }} is a ${TEAM} team member"
+            echo "::set-output name=message::true"
+          else
+            echo "Actor ${{ github.actor }} is NOT a ${TEAM} team member"
+            echo "::set-output name=message::false"
+          fi
+        env:
+          GITHUB_TOKEN: ${{ secrets.JIRA_SYNC_GITHUB_TOKEN }}
+
+      - name: Create ticket
+        if: ( github.event.action == 'opened' && steps.set-ticket-type.outputs.type != 'Task' ) || ( github.event.action == 'opened' && steps.set-ticket-type.outputs.type == 'Task' && steps.is-team-member.outputs.message == 'false' )
```

Comment:
Just want to make sure I understand the intent behind `is-team-member`. This keeps JIRA from auto generating tickets we create in GitHub right?

Reply:
Yes, that is correct, this is mostly a copy and pasted from what's in https://github.com/hashicorp/hcdiag/blob/main/.github/workflows/jira.yml

Reply:
ok cool, all good!